### ISSUE TITLE
Ability to disable DrawerNavigationItem items

### ISSUE THREAD

**Issue by siansell** [exponentjs/ex-navigation#116]
Hi,

First, thanks for and great job with exponent - it's really made it a lot more fun getting started with react-native development, particularly never having to open Xcode!

I would like the ability to disable the onPress and onLongPress events of a DrawerNavigationItem. For example:

```
<DrawerNavigationItem
  id='auth'
  disabled={true}
  renderTitle={() => this._renderAuthTitle()}
  showsTouches={false}
/>
```

It would be as simple as changing lines 115-130 of https://github.com/exponentjs/ex-navigation/blob/master/src/drawer/ExNavigationDrawerLayout.js to:

```
_handlePress = (item: any) => {
  if (item.disabled) return;
  item.onPress();
  this._component.closeDrawer();
}

_handleLongPress = (item: any) => {
  if (!item.onLongPress || item.disabled) {
    return;
  }

  item.onLongPress();
  this._component.closeDrawer();
}
```

My use case is that sometimes I want to create custom DrawerNavigationItems that contain multiple onPress handlers, for example my `renderAuthTitle()` above is "Sign In or Register", which pushes the relevant route depending on whether "Sign In" or "Register" was clicked. Or sometimes I want to create purely presentational DrawerNavigationItems.

I'm happy to send a PR (even though I've never sent one before!) if you think this is a reasonable idea.

**Comment by chirag04** [comment]
@siansell what about passing `onPress` as an empty function? It will still close the drawer tho.

**Comment by siansell** [comment]
@chirag04 Yeah, that was the first thing I tried but still closing the door is not desirable. The DrawerNavigationItem takes the full width of the drawer so the user can press any where outside of the intended targets and the drawer will close.

**Comment by siansell** [comment]
An alternative would be to make the `onPress()` and `onLongPress()` handling consistent, for example:

```
_handlePress = (item: any) => {
  if (!item.onPress) {
    return;
  }

  item.onPress();
  this._component.closeDrawer();
}

_handleLongPress = (item: any) => {
  if (!item.onLongPress) {
    return;
  }

  item.onLongPress();
  this._component.closeDrawer();
}
```

What do people think?

**Comment by satya164** [comment]
Probably you can just use `renderHeader` to achieve what you're doing?

**Comment by siansell** [comment]
@satya164 Actually you're right, in my case I can use `renderHeader` because in this case the relevant `DrawerNavigationItem`happens to be at the top - for some reason I didn't think of that.

There could be cases where I want more control over other DrawerNavigationItems, though.

**Comment by satya164** [comment]
Yeah, the `disabled` prop is useful

**Comment by JonasJonny** [comment]
+1 for PR.
